common/docserver/docserver.go

```diff
@@ -95,14 +95,20 @@ func (self *DocServer) Get(uri, path string) (content []byte, err error) {
 			resp.Body.Close()
 		}
 	}()
+
 	if err != nil {
 		return
 	}
+
 	content, err = ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return
 	}
 
+	if resp.StatusCode >= 400 {
```

Comment:
I'm not sure if this is valid. There are some other error codes which are < 400 (e.g. 304) which can be returned and don't contain the actual requested data. But maybe these are never returned in this case?